tests/Emulation/Authenticator.hs

```diff
@@ -459,11 +460,28 @@ newKeyPair PublicKey.COSEAlgorithmIdentifierEdDSA = do
   secret <- Ed25519.generateSecretKey
   let public = Ed25519.toPublic secret
   pure (PublicKey.Ed25519PublicKey public, PrivateKey.Ed25519PrivateKey secret)
+newKeyPair PublicKey.COSEAlgorithmIdentifierRS1 = newRSAKeyPair PublicKey.COSEAlgorithmIdentifierRS1 160
+newKeyPair PublicKey.COSEAlgorithmIdentifierRS256 = newRSAKeyPair PublicKey.COSEAlgorithmIdentifierRS256 256
+newKeyPair PublicKey.COSEAlgorithmIdentifierRS384 = newRSAKeyPair PublicKey.COSEAlgorithmIdentifierRS384 384
+newKeyPair PublicKey.COSEAlgorithmIdentifierRS512 = newRSAKeyPair PublicKey.COSEAlgorithmIdentifierRS512 512
 
 newECDSAKeyPair :: MonadRandom m => PublicKey.COSEAlgorithmIdentifier -> m (PublicKey.PublicKey, PrivateKey.PrivateKey)
 newECDSAKeyPair ident = do
   let curve = ECC.getCurveByName $ PublicKey.toCurveName ident
   (public, private) <- ECC.generate curve
-  let privateKey = fromMaybe (error "Not a ECDSAKey") $ PrivateKey.toECDSAKey ident private
-      publicKey = fromMaybe (error "Not a ECDSAKey") $ PublicKey.toECDSAKey ident public
+  let privateKey = fromMaybe (error "Not an ECDSAKey") $ PrivateKey.toECDSAKey ident private
+      publicKey = fromMaybe (error "Not an ECDSAKey") $ PublicKey.toECDSAKey ident public
   pure (publicKey, privateKey)
+
+newRSAKeyPair :: MonadRandom m => PublicKey.COSEAlgorithmIdentifier -> Int -> m (PublicKey.PublicKey, PrivateKey.PrivateKey)
+newRSAKeyPair ident publicSize = do
+  (public, private) <- RSA.generate publicSize 65537
```

Comment:
Where does this constant come from?

Reply:
Convention, this is one of the two most frequently used exponents, and the one deemed most secure. I can't really link a spec here, but I will add a comment.